app/user.go

```diff
@@ -1,36 +1,78 @@
 package app
 
 import (
+	"encoding/json"
 	"errors"
+	"os"
 
-	"github.com/ubclaunchpad/cumulus/consensus"
+	crand "crypto/rand"
 
 	"github.com/ubclaunchpad/cumulus/blockchain"
+	"github.com/ubclaunchpad/cumulus/consensus"
 	"github.com/ubclaunchpad/cumulus/msg"
-
-	crand "crypto/rand"
 )
 
 // User holds basic user information.
 type User struct {
-	blockchain.Wallet
+	Wallet    *blockchain.Wallet
 	Name      string
 	BlockSize uint32
 }
 
 // NewUser creates a new user
 func NewUser() *User {
 	return &User{
-		Wallet:    *blockchain.NewWallet(),
+		Wallet:    blockchain.NewWallet(),
 		BlockSize: blockchain.DefaultBlockSize,
+		Name:      "Default User",
+	}
+}
+
+// Public returns the public key (address) of the given user
+func (u *User) Public() blockchain.Address {
+	return u.Wallet.Public()
+}
+
+// Save writes the user to a file of the given name in the current working
+// directory in JSON format. It returns an error if one occurred, or a pointer
+// to the file that was written to on success.
+func (u *User) Save(fileName string) error {
+	userFile, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE, 0775)
+	if err != nil {
+		return err
 	}
+	defer userFile.Close()
+
+	userBytes, err := json.Marshal(u)
+	if err != nil {
+		return err
+	}
+
+	if _, err = userFile.Write(userBytes); err != nil {
+		return err
+	}
+	return nil
 }
 
-// getCurrentUser gets the current user function only used for app initalization.
-func getCurrentUser() *User {
-	// TODO: Check for local user information on disk,
-	// If doesnt exist, create new user.
-	return NewUser()
+// Load attempts to read user info from the file with the given name in the
+// current working directory in JSON format. On success this returns
+// a pointer to a new user constructed from the information in the file.
+// If an error occurrs it is returned.
+func Load(fileName string) (*User, error) {
```

Comment:
I'd call this `LoadUser` since it's in the `app` namespace. `app.Load` seems like it should load the app

Reply:
Will do.